### DISCUSSION TITLE
Upgrading from 3.4 -> 4.4, Composer uninstalls all Symfony packages and Flex deletes everything

### DISCUSSION THREAD

**Discussion by Bilge** [symfony/symfony#43966] [Q&A] [answered]
I have successfully upgraded Symfony applications in the recent past, including this one, but after updating my package constraints from `3.4.*` -> `4.4.*`, Composer uninstalled almost everything to do with Symfony, except `symfony/symfony` and Flex removed half my config files, including my `Kernel` and entry points like `index.php`. However, I do not require `symfony/symfony`, I specify the components I want manually, including `symfony/framework-bundle`, which is missing in lieu of `symfony/symfony`.

Running `composer why symfony/symfony` gives the following:

>nelmio/api-doc-bundle      2.13.5       conflicts  symfony/symfony (~2.7.8)
roave/security-advisories  9999999-dev  conflicts  symfony/symfony (>=2,<3.4.49|>=4,<4.4.24|>=5,<5.2.9|>=5.3,<5.3.2)
symfony/symfony            v4.4.33      replaces   symfony/amazon-mailer (self.version)
symfony/symfony            v4.4.33      replaces   symfony/asset (self.version)
symfony/symfony            v4.4.33      replaces   symfony/browser-kit (self.version)
symfony/symfony            v4.4.33      replaces   symfony/cache (self.version)
symfony/symfony            v4.4.33      replaces   symfony/config (self.version)
symfony/symfony            v4.4.33      replaces   symfony/console (self.version)
symfony/symfony            v4.4.33      replaces   symfony/css-selector (self.version)
symfony/symfony            v4.4.33      replaces   symfony/debug (self.version)
symfony/symfony            v4.4.33      replaces   symfony/debug-bundle (self.version)
symfony/symfony            v4.4.33      replaces   symfony/dependency-injection (self.version)
symfony/symfony            v4.4.33      replaces   symfony/doctrine-bridge (self.version)
symfony/symfony            v4.4.33      replaces   symfony/dom-crawler (self.version)
symfony/symfony            v4.4.33      replaces   symfony/dotenv (self.version)
symfony/symfony            v4.4.33      replaces   symfony/error-handler (self.version)
symfony/symfony            v4.4.33      replaces   symfony/event-dispatcher (self.version)
symfony/symfony            v4.4.33      replaces   symfony/expression-language (self.version)
symfony/symfony            v4.4.33      replaces   symfony/filesystem (self.version)
symfony/symfony            v4.4.33      replaces   symfony/finder (self.version)
symfony/symfony            v4.4.33      replaces   symfony/form (self.version)
symfony/symfony            v4.4.33      replaces   symfony/framework-bundle (self.version)
symfony/symfony            v4.4.33      replaces   symfony/google-mailer (self.version)
symfony/symfony            v4.4.33      replaces   symfony/http-client (self.version)
symfony/symfony            v4.4.33      replaces   symfony/http-foundation (self.version)
symfony/symfony            v4.4.33      replaces   symfony/http-kernel (self.version)
symfony/symfony            v4.4.33      replaces   symfony/inflector (self.version)
symfony/symfony            v4.4.33      replaces   symfony/intl (self.version)
symfony/symfony            v4.4.33      replaces   symfony/ldap (self.version)
symfony/symfony            v4.4.33      replaces   symfony/lock (self.version)
symfony/symfony            v4.4.33      replaces   symfony/mailchimp-mailer (self.version)
symfony/symfony            v4.4.33      replaces   symfony/mailer (self.version)
symfony/symfony            v4.4.33      replaces   symfony/mailgun-mailer (self.version)
symfony/symfony            v4.4.33      replaces   symfony/messenger (self.version)
symfony/symfony            v4.4.33      replaces   symfony/mime (self.version)
symfony/symfony            v4.4.33      replaces   symfony/monolog-bridge (self.version)
symfony/symfony            v4.4.33      replaces   symfony/options-resolver (self.version)
symfony/symfony            v4.4.33      replaces   symfony/postmark-mailer (self.version)
symfony/symfony            v4.4.33      replaces   symfony/process (self.version)
symfony/symfony            v4.4.33      replaces   symfony/property-access (self.version)
symfony/symfony            v4.4.33      replaces   symfony/property-info (self.version)
symfony/symfony            v4.4.33      replaces   symfony/proxy-manager-bridge (self.version)
symfony/symfony            v4.4.33      replaces   symfony/routing (self.version)
symfony/symfony            v4.4.33      replaces   symfony/security (self.version)
symfony/symfony            v4.4.33      replaces   symfony/security-bundle (self.version)
symfony/symfony            v4.4.33      replaces   symfony/security-core (self.version)
symfony/symfony            v4.4.33      replaces   symfony/security-csrf (self.version)
symfony/symfony            v4.4.33      replaces   symfony/security-guard (self.version)
symfony/symfony            v4.4.33      replaces   symfony/security-http (self.version)
symfony/symfony            v4.4.33      replaces   symfony/sendgrid-mailer (self.version)
symfony/symfony            v4.4.33      replaces   symfony/serializer (self.version)
symfony/symfony            v4.4.33      replaces   symfony/stopwatch (self.version)
symfony/symfony            v4.4.33      replaces   symfony/templating (self.version)
symfony/symfony            v4.4.33      replaces   symfony/translation (self.version)
symfony/symfony            v4.4.33      replaces   symfony/twig-bridge (self.version)
symfony/symfony            v4.4.33      replaces   symfony/twig-bundle (self.version)
symfony/symfony            v4.4.33      replaces   symfony/validator (self.version)
symfony/symfony            v4.4.33      replaces   symfony/var-dumper (self.version)
symfony/symfony            v4.4.33      replaces   symfony/var-exporter (self.version)
symfony/symfony            v4.4.33      replaces   symfony/web-link (self.version)
symfony/symfony            v4.4.33      replaces   symfony/web-profiler-bundle (self.version)
symfony/symfony            v4.4.33      replaces   symfony/web-server-bundle (self.version)
symfony/symfony            v4.4.33      replaces   symfony/workflow (self.version)
symfony/symfony            v4.4.33      replaces   symfony/yaml (self.version)

I don't understand this output: it does not tell anything about why it installed the `symfony/symfony` package. It certainly is not in my root `composer.json`. Removing either `nelmio/api-doc-bundle` or `roave/security-advisories` does not change anything.

I also tried running `composer why-not symfony/framework-bundle 4.4.31` which gives:
>There is no installed package depending on "symfony/framework-bundle" in versions not matching 4.4.31

I am at a loss. I don't know why this is happening or what to do.

**Comment by stof** [comment]
Well, if `symfony/symfony` is in your vendor folder, it is totally expected that composer will remove the `symfony/*` subtree packages, as it does not make sense to install both the fullstack repo and the subtree repos.

Note that the Flex skeleton has a conflict rule forbidding the fullstack repo, as it is incompatible with recipes (and with smart defaults based on enabling installed packages). So I have even less idea why it would be installed.

**Comment by Bilge** [comment, marked as answer]
If I manually remove (`rm -rf vendor/symfony/symfony`), then run `composer u`, it reinstalls `symfony/symfony`, so it seems convinced by something that it really needs to install it.

**Reply by stof** [reply]
Search for `symfony/symfony` in the lock file, to find what could be requiring it.

Also, do you have the conflict rule ? It might workaround the issue.

**Reply by Bilge** [reply]
I did not have the conflict rule since it's a Symfony 2 project originally. I copied it from another project

```json
    "conflict": {
        "symfony/symfony": "*"
    },
```

Doing so, I can see the culprit when doing `composer u`.

>hwi/oauth-bundle[0.6.0, ..., 0.6.3] require php-http/client-implementation ^1.0 -> satisfiable by symfony/symfony[v4.4.20, ..., v4.4.33].

I remain unclear why this is not presented when doing `composer why`.

**Reply by stof** [reply]
this is also satisfiable by symfony/http-client though

**Reply by Bilge** [reply]
Manually installing `symfony/http-client` seems to resolve the issue 👍🏻
